kusto/ingest/managed.go

```diff
@@ -44,11 +46,72 @@ func NewManaged(client QueryClient, db, table string, options ...Option) (*Manag
 	}, nil
 }
 
+// Attempts to stream with retries, on success - return res,nil.
+// If failed permanently - return err,nil.
+// If failed transiently - return nil,nil.
+func (m *Managed) streamWithRetries(ctx context.Context, payloadProvider func() io.Reader, props properties.All, isBlobUri bool) (*Result, error) {
```

Comment:
Why is payloadprovider a function rather than just an argument?

Reply:
Also do you close it? need to close it?

Reply:
See line 166 
	res, err := m.streamWithRetries(ctx, func() io.Reader { return bytes.NewReader(buf) }, props, false)

Reader interface has only Read() no close ...